### ISSUE TITLE
Metrics/LineLength AllowURI does not work with interpolated strings

### ISSUE THREAD

**Issue by michaeljb** [bbatsov/rubocop#1503]
If I have a string with a URI that is over the `Metrics/LineLength` limit, with parts of it defined by variables that are interpolated, the line with that URI causes a violation, even with `AllowURI` set to `true`.

```yaml
# example.yaml
Metrics/LineLength:
  Max: 20
  AllowURI: true  
```
```ruby
# example.rb
def example
  foo = 'bar'
  "https://www.github.com/michaeljb/#{foo}"
end
```
```text
$ rubocop --config=example.yaml example.rb
Inspecting 1 file
C

Offenses:

example.rb:3:38: C: Line is too long. [43/20]
  "https://www.github.com/michaeljb/#{foo}"
                                     ^^^^^^

1 file inspected, 1 offense detected
```

I believe the problem is with the definition of [`LineLength#valid_url?`](https://github.com/bbatsov/rubocop/blob/v0.28.0/lib/rubocop/cop/metrics/line_length.rb#L76), which uses `URI.parse` to make sure the violating line really is a URI, and it cannot handle curly braces:

```text
irb(main):001:0> require 'uri'
=> true
irb(main):002:0> URI.parse('http://www.#{foo}.com')
URI::InvalidURIError: bad URI(is not URI?): http://www.#{foo}.com
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:176:in `split'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:211:in `parse'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:747:in `parse'
	from (irb):2
	from /usr/bin/irb:12:in `<main>'```
